src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.Extension/syntaxes/aspnetcorerazor.tmLanguage.yml

```diff
@@ -641,6 +642,20 @@ repository:
     endCaptures:
       1: { name: 'punctuation.curlybrace.close.cs' }
 
+  #>>>>> @lock (...) { ... } <<<<<
+
+  lock-statement:
+    name: 'meta.statement.lock.razor'
+    begin: '(@)(lock)\b\s*(?=\()'
+    beginCaptures:
+      1: { patterns: [ include: '#transition' ] }
+      2: { name: 'keyword.other.lock.cs' }
+    patterns:
+      - include: '#csharp-condition'
```

Comment:
We talked offline but I'll call this out for others: this is catching the somevar portion of `lock (somevar) {...}`. It's not pretty, but it works and it's what the csharp grammar does, and we try to use their stuff as much as possible.